packages/cli-repl/src/cli-repl.ts

```diff
@@ -310,10 +311,14 @@ class CliRepl {
    * The greeting for the shell.
    */
   greet(): void {
-    const { version } = require('../package.json');
+    const { version, engines } = require('../package.json');
     console.log(`Using MongoDB: ${this.internalState.connectionInfo.buildInfo.version}`);
     console.log(`${clr('Using Mongosh Beta', ['bold', 'yellow'])}: ${version}`);
     console.log(`${MONGOSH_WIKI}`);
+    if (semver.lt(process.version, engines.node.replace('^', ''))) {
```

Comment:
Very very minor. What do you think having a function `verifyNodeEngineVersion` either apart from `greet` or used inside `greet`? I feel it would be easier for a reader to know about that functionality, but I may be wrong. :)

Reply:
+1, I'd probably have this either in constructor or setupRepl before we connect to mongodb. I think if we do this after we connect to mongodb, we should make sure we close the connection as well.

Reply:
Ok ty!

Reply:
done